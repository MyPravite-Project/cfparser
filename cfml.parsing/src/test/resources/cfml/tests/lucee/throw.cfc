component{
  function foo(){
    throw message="";
  }
}