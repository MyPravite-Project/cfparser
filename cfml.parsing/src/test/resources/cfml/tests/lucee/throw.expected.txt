/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <foo>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
THROW                    <throw>
IDENTIFIER               <message>
'='                      <=>
OPEN_STRING              <">
CLOSE_STRING             <">
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier foo)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (tagOperatorStatement
                (tagThrowStatement
                  throw
                  (paramStatementAttributes
                    (param
                      (multipartIdentifier (identifier message))
                      =
                      (startExpression
                        (baseExpression
                          (unaryExpression (primaryExpression (literalExpression (stringLiteral " "))))
                        )
                      )
                    )
                  )
                )
              )
            )
            (statement ;)
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function foo()     {
      throw  message='';

    }
}
/*======*/